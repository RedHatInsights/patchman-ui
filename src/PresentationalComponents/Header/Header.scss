.pf-v5-c-tabs.patchman-tabs {
    background-color: var(--pf-v5-global--BackgroundColor--100);
    padding-left: var(--pf-v5-c-page__main-section--md--PaddingLeft);
    display: block;
}
