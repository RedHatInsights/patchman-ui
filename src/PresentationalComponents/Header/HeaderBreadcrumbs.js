import {
    Breadcrumb,
    BreadcrumbItem
} from '@patternfly/react-core/dist/js/components/Breadcrumb/';
import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

const HeaderBreadcrumbs = ({ items, headerOUIA }) => {
    return (
        <Breadcrumb>
            {items.filter(Boolean).map(item => (
                <BreadcrumbItem key={item.title} isActive={item.isActive}>
                    {(item.to && <Link to={item.to}
                        data-ouia-component-type={`${headerOUIA}-breadcrumb`}
                        data-ouia-component-id={`breadcrumb-to-${item.title}`}
                    >{item.title}</Link>) ||
                        item.title}
                </BreadcrumbItem>
            ))}
        </Breadcrumb>
    );
};

HeaderBreadcrumbs.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({
            isActive: PropTypes.bool,
            to: PropTypes.string,
            title: PropTypes.string
        })
    ),
    headerOUIA: PropTypes.string
};

export default HeaderBreadcrumbs;
