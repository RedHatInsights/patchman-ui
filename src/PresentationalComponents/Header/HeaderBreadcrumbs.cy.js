import React from 'react';
import { mount } from '@cypress/react';

import HeaderBreadcrumbs from './HeaderBreadcrumbs';
import { MemoryRouter } from 'react-router-dom';

describe('HeaderBreadcrumbs', () => {
    it('renders two breadcrumbs', () => {
        mount(
            <HeaderBreadcrumbs
                items={[
                    {
                        title: 'foobar',
                        isActive: false
                    },
                    {
                        title: 'foobar last',
                        isActive: true
                    }
                ]}
            />
        );
        cy.get('.pf-c-breadcrumb__list').children().should('have.length', 2);
    });

    it('renders one breadcrumb with link', () => {
        mount(
            // Router wrapper is needed since one of the child components render Route
            <MemoryRouter>
                <HeaderBreadcrumbs
                    items={[
                        {
                            title: 'foobar',
                            isActive: false,
                            to: 'test'
                        },
                        {
                            title: 'foobar last',
                            isActive: true
                        }
                    ]}
                />
            </MemoryRouter>
        );
        cy.get('.pf-c-breadcrumb__list').children().should('have.length', 2);
        cy.get('.pf-c-breadcrumb__list')
        .children()
        .eq(0)
        .find('a')
        .should('have.attr', 'href', '/test');
    });
});
