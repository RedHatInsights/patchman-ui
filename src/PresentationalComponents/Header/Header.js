import {
    PageHeader,
    PageHeaderTitle
} from '@redhat-cloud-services/frontend-components/components/PageHeader';
import PropTypes from 'prop-types';
import React from 'react';
import HeaderBreadcrumbs from './HeaderBreadcrumbs';
import HeaderTabs from './HeaderTabs';

const Header = ({ title, showTabs, breadcrumbs, children, headerOUIA }) => {
    return (
        <React.Fragment>
            <PageHeader
                data-ouia-component-type={`${headerOUIA}-page-header`}
            >
                {breadcrumbs && <HeaderBreadcrumbs items={breadcrumbs} headerOUIA={headerOUIA} />}
                <PageHeaderTitle title={title} />
                {children}
            </PageHeader>
            {showTabs && <HeaderTabs headerOUIA = {headerOUIA}/>}
        </React.Fragment>
    );
};

Header.propTypes = {
    title: PropTypes.string,
    showTabs: PropTypes.bool,
    breadcrumbs: PropTypes.array,
    children: PropTypes.any,
    headerOUIA: PropTypes.string
};

export default Header;
