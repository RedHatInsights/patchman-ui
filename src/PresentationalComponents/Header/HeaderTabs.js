import { Tab, Tabs } from '@patternfly/react-core/dist/js/components/Tabs';
import propTypes from 'prop-types';
import React from 'react';
import { withRouter } from 'react-router-dom';
import { paths } from '../../Routes';
import './Header.scss';

const HeaderTabs = ({ history, headerOUIA }) => {
    const handleRedirect = (event, tabString) => {
        history.push(tabString);
    };

    return (
        <Tabs
            onSelect={handleRedirect}
            activeKey={history.location.pathname}
            className={'patchman-tabs'}
        >
            <Tab
                eventKey={paths.advisories.to}
                title={paths.advisories.title}
                data-ouia-component-type={`${headerOUIA}-tab`}
                data-ouia-component-id={`${headerOUIA}-tab-${paths.advisories.title}`}
            />
            <Tab
                eventKey={paths.systems.to}
                title={paths.systems.title}
                data-ouia-component-type={`${headerOUIA}-tab`}
                data-ouia-component-id={`${headerOUIA}-tab-${paths.systems.title}`}
            />
        </Tabs>
    );
};

HeaderTabs.propTypes = {
    history: propTypes.object,
    headerOUIA: propTypes.string
};

export default withRouter(HeaderTabs);
