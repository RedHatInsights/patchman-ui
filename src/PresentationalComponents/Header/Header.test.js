/* eslint-disable */
import toJson from 'enzyme-to-json';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from './Header';

const withLink = Component => {
    return mount(<Router>{Component}</Router>);
};
describe('Header component', () => {
    it('should render with header as empty string', () => {
        const header = { title: '' };
        const wrapper = withLink(<Header {...header} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with header Hello world', () => {
        const header = { title: 'Hello world' };
        const wrapper = withLink(<Header {...header} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with empty breadcrumb', () => {
        const header = {
            breadcrumbs: [],
            title: ''
        };
        const wrapper = withLink(<Header {...header} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with 1 breadcrumb item and last is active', () => {
        const header = {
            title: '',
            breadcrumbs: [
                {
                    title: 'First item',
                    to: '#',
                    isActive: true
                }
            ]
        };
        const wrapper = withLink(<Header {...header} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render without to attribute', () => {
        const header = {
            title: '',
            breadcrumbs: [
                {
                    title: 'First item',
                    isActive: true
                }
            ]
        };
        const wrapper = withLink(<Header {...header} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with 2 breadcrumb items and last is active', () => {
        const header = {
            title: '',
            breadcrumbs: [
                {
                    title: 'First item',
                    to: '#',
                    isActive: false
                },
                {
                    title: 'Second item',
                    to: '#',
                    isActive: true
                }
            ]
        };
        const wrapper = withLink(<Header {...header} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with 3 breadcrumb items and last is active', () => {
        const header = {
            showTabs: false,
            title: '',
            breadcrumbs: [
                {
                    title: 'First item',
                    to: '#',
                    isActive: false
                },
                {
                    title: 'Second item',
                    to: '#',
                    isActive: false
                },
                {
                    title: 'Third item',
                    to: '#',
                    isActive: true
                }
            ]
        };
        const wrapper = withLink(<Header {...header} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with tabs only, click a tab', () => {
        const wrapper = withLink(<Header showTabs title="Title" />);
        let history = wrapper.find('Router').props().history;
        const spy = jest.spyOn(history, 'push');
        wrapper
            .find('Tabs.patchman-tabs')
            .props()
            .onSelect();

        expect(spy).toBeCalled();
    });
});
/* eslint-enable */
