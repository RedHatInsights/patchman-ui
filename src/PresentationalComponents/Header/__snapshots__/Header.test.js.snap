// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render with 1 breadcrumb item and last is active 1`] = `
<StaticRouter
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Header
      breadcrumbs={
        Array [
          Object {
            "isActive": true,
            "title": "First item",
            "to": "#",
          },
        ]
      }
      title=""
    >
      <PageHeader
        data-ouia-component-type="undefined-page-header"
      >
        <section
          className="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
          data-ouia-component-type="undefined-page-header"
          widget-type="InsightsPageHeader"
        >
          <HeaderBreadcrumbs
            items={
              Array [
                Object {
                  "isActive": true,
                  "title": "First item",
                  "to": "#",
                },
              ]
            }
          >
            <Breadcrumb>
              <nav
                aria-label="Breadcrumb"
                className="pf-c-breadcrumb"
                data-ouia-component-id={1}
                data-ouia-component-type="PF4/Breadcrumb"
                data-ouia-safe={true}
              >
                <ol
                  className="pf-c-breadcrumb__list"
                >
                  <BreadcrumbItem
                    isActive={true}
                    key=".$First item"
                    showDivider={false}
                  >
                    <li
                      className="pf-c-breadcrumb__item"
                    >
                      <Link
                        data-ouia-component-id="breadcrumb-to-First item"
                        data-ouia-component-type="undefined-breadcrumb"
                        to="#"
                      >
                        <LinkAnchor
                          data-ouia-component-id="breadcrumb-to-First item"
                          data-ouia-component-type="undefined-breadcrumb"
                          href="/"
                          navigate={[Function]}
                        >
                          <a
                            data-ouia-component-id="breadcrumb-to-First item"
                            data-ouia-component-type="undefined-breadcrumb"
                            href="/"
                            onClick={[Function]}
                          >
                            First item
                          </a>
                        </LinkAnchor>
                      </Link>
                    </li>
                  </BreadcrumbItem>
                </ol>
              </nav>
            </Breadcrumb>
          </HeaderBreadcrumbs>
          <PageHeaderTitle
            title=""
          >
            <Title
              className=""
              headingLevel="h1"
              size="2xl"
              widget-type="InsightsPageHeaderTitle"
            >
              <h1
                className="pf-c-title pf-m-2xl"
                widget-type="InsightsPageHeaderTitle"
              >
                 
                 
              </h1>
            </Title>
          </PageHeaderTitle>
        </section>
      </PageHeader>
    </Header>
  </Router>
</StaticRouter>
`;

exports[`Header component should render with 2 breadcrumb items and last is active 1`] = `
<StaticRouter
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Header
      breadcrumbs={
        Array [
          Object {
            "isActive": false,
            "title": "First item",
            "to": "#",
          },
          Object {
            "isActive": true,
            "title": "Second item",
            "to": "#",
          },
        ]
      }
      title=""
    >
      <PageHeader
        data-ouia-component-type="undefined-page-header"
      >
        <section
          className="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
          data-ouia-component-type="undefined-page-header"
          widget-type="InsightsPageHeader"
        >
          <HeaderBreadcrumbs
            items={
              Array [
                Object {
                  "isActive": false,
                  "title": "First item",
                  "to": "#",
                },
                Object {
                  "isActive": true,
                  "title": "Second item",
                  "to": "#",
                },
              ]
            }
          >
            <Breadcrumb>
              <nav
                aria-label="Breadcrumb"
                className="pf-c-breadcrumb"
                data-ouia-component-id={3}
                data-ouia-component-type="PF4/Breadcrumb"
                data-ouia-safe={true}
              >
                <ol
                  className="pf-c-breadcrumb__list"
                >
                  <BreadcrumbItem
                    isActive={false}
                    key=".$First item"
                    showDivider={false}
                  >
                    <li
                      className="pf-c-breadcrumb__item"
                    >
                      <Link
                        data-ouia-component-id="breadcrumb-to-First item"
                        data-ouia-component-type="undefined-breadcrumb"
                        to="#"
                      >
                        <LinkAnchor
                          data-ouia-component-id="breadcrumb-to-First item"
                          data-ouia-component-type="undefined-breadcrumb"
                          href="/"
                          navigate={[Function]}
                        >
                          <a
                            data-ouia-component-id="breadcrumb-to-First item"
                            data-ouia-component-type="undefined-breadcrumb"
                            href="/"
                            onClick={[Function]}
                          >
                            First item
                          </a>
                        </LinkAnchor>
                      </Link>
                    </li>
                  </BreadcrumbItem>
                  <BreadcrumbItem
                    isActive={true}
                    key=".$Second item"
                    showDivider={true}
                  >
                    <li
                      className="pf-c-breadcrumb__item"
                    >
                      <span
                        className="pf-c-breadcrumb__item-divider"
                      >
                        <AngleRightIcon
                          color="currentColor"
                          noVerticalAlign={false}
                          size="sm"
                        >
                          <svg
                            aria-hidden={true}
                            aria-labelledby={null}
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style={
                              Object {
                                "verticalAlign": "-0.125em",
                              }
                            }
                            viewBox="0 0 256 512"
                            width="1em"
                          >
                            <path
                              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                              transform=""
                            />
                          </svg>
                        </AngleRightIcon>
                      </span>
                      <Link
                        data-ouia-component-id="breadcrumb-to-Second item"
                        data-ouia-component-type="undefined-breadcrumb"
                        to="#"
                      >
                        <LinkAnchor
                          data-ouia-component-id="breadcrumb-to-Second item"
                          data-ouia-component-type="undefined-breadcrumb"
                          href="/"
                          navigate={[Function]}
                        >
                          <a
                            data-ouia-component-id="breadcrumb-to-Second item"
                            data-ouia-component-type="undefined-breadcrumb"
                            href="/"
                            onClick={[Function]}
                          >
                            Second item
                          </a>
                        </LinkAnchor>
                      </Link>
                    </li>
                  </BreadcrumbItem>
                </ol>
              </nav>
            </Breadcrumb>
          </HeaderBreadcrumbs>
          <PageHeaderTitle
            title=""
          >
            <Title
              className=""
              headingLevel="h1"
              size="2xl"
              widget-type="InsightsPageHeaderTitle"
            >
              <h1
                className="pf-c-title pf-m-2xl"
                widget-type="InsightsPageHeaderTitle"
              >
                 
                 
              </h1>
            </Title>
          </PageHeaderTitle>
        </section>
      </PageHeader>
    </Header>
  </Router>
</StaticRouter>
`;

exports[`Header component should render with 3 breadcrumb items and last is active 1`] = `
<StaticRouter
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Header
      breadcrumbs={
        Array [
          Object {
            "isActive": false,
            "title": "First item",
            "to": "#",
          },
          Object {
            "isActive": false,
            "title": "Second item",
            "to": "#",
          },
          Object {
            "isActive": true,
            "title": "Third item",
            "to": "#",
          },
        ]
      }
      showTabs={false}
      title=""
    >
      <PageHeader
        data-ouia-component-type="undefined-page-header"
      >
        <section
          className="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
          data-ouia-component-type="undefined-page-header"
          widget-type="InsightsPageHeader"
        >
          <HeaderBreadcrumbs
            items={
              Array [
                Object {
                  "isActive": false,
                  "title": "First item",
                  "to": "#",
                },
                Object {
                  "isActive": false,
                  "title": "Second item",
                  "to": "#",
                },
                Object {
                  "isActive": true,
                  "title": "Third item",
                  "to": "#",
                },
              ]
            }
          >
            <Breadcrumb>
              <nav
                aria-label="Breadcrumb"
                className="pf-c-breadcrumb"
                data-ouia-component-id={4}
                data-ouia-component-type="PF4/Breadcrumb"
                data-ouia-safe={true}
              >
                <ol
                  className="pf-c-breadcrumb__list"
                >
                  <BreadcrumbItem
                    isActive={false}
                    key=".$First item"
                    showDivider={false}
                  >
                    <li
                      className="pf-c-breadcrumb__item"
                    >
                      <Link
                        data-ouia-component-id="breadcrumb-to-First item"
                        data-ouia-component-type="undefined-breadcrumb"
                        to="#"
                      >
                        <LinkAnchor
                          data-ouia-component-id="breadcrumb-to-First item"
                          data-ouia-component-type="undefined-breadcrumb"
                          href="/"
                          navigate={[Function]}
                        >
                          <a
                            data-ouia-component-id="breadcrumb-to-First item"
                            data-ouia-component-type="undefined-breadcrumb"
                            href="/"
                            onClick={[Function]}
                          >
                            First item
                          </a>
                        </LinkAnchor>
                      </Link>
                    </li>
                  </BreadcrumbItem>
                  <BreadcrumbItem
                    isActive={false}
                    key=".$Second item"
                    showDivider={true}
                  >
                    <li
                      className="pf-c-breadcrumb__item"
                    >
                      <span
                        className="pf-c-breadcrumb__item-divider"
                      >
                        <AngleRightIcon
                          color="currentColor"
                          noVerticalAlign={false}
                          size="sm"
                        >
                          <svg
                            aria-hidden={true}
                            aria-labelledby={null}
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style={
                              Object {
                                "verticalAlign": "-0.125em",
                              }
                            }
                            viewBox="0 0 256 512"
                            width="1em"
                          >
                            <path
                              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                              transform=""
                            />
                          </svg>
                        </AngleRightIcon>
                      </span>
                      <Link
                        data-ouia-component-id="breadcrumb-to-Second item"
                        data-ouia-component-type="undefined-breadcrumb"
                        to="#"
                      >
                        <LinkAnchor
                          data-ouia-component-id="breadcrumb-to-Second item"
                          data-ouia-component-type="undefined-breadcrumb"
                          href="/"
                          navigate={[Function]}
                        >
                          <a
                            data-ouia-component-id="breadcrumb-to-Second item"
                            data-ouia-component-type="undefined-breadcrumb"
                            href="/"
                            onClick={[Function]}
                          >
                            Second item
                          </a>
                        </LinkAnchor>
                      </Link>
                    </li>
                  </BreadcrumbItem>
                  <BreadcrumbItem
                    isActive={true}
                    key=".$Third item"
                    showDivider={true}
                  >
                    <li
                      className="pf-c-breadcrumb__item"
                    >
                      <span
                        className="pf-c-breadcrumb__item-divider"
                      >
                        <AngleRightIcon
                          color="currentColor"
                          noVerticalAlign={false}
                          size="sm"
                        >
                          <svg
                            aria-hidden={true}
                            aria-labelledby={null}
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style={
                              Object {
                                "verticalAlign": "-0.125em",
                              }
                            }
                            viewBox="0 0 256 512"
                            width="1em"
                          >
                            <path
                              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                              transform=""
                            />
                          </svg>
                        </AngleRightIcon>
                      </span>
                      <Link
                        data-ouia-component-id="breadcrumb-to-Third item"
                        data-ouia-component-type="undefined-breadcrumb"
                        to="#"
                      >
                        <LinkAnchor
                          data-ouia-component-id="breadcrumb-to-Third item"
                          data-ouia-component-type="undefined-breadcrumb"
                          href="/"
                          navigate={[Function]}
                        >
                          <a
                            data-ouia-component-id="breadcrumb-to-Third item"
                            data-ouia-component-type="undefined-breadcrumb"
                            href="/"
                            onClick={[Function]}
                          >
                            Third item
                          </a>
                        </LinkAnchor>
                      </Link>
                    </li>
                  </BreadcrumbItem>
                </ol>
              </nav>
            </Breadcrumb>
          </HeaderBreadcrumbs>
          <PageHeaderTitle
            title=""
          >
            <Title
              className=""
              headingLevel="h1"
              size="2xl"
              widget-type="InsightsPageHeaderTitle"
            >
              <h1
                className="pf-c-title pf-m-2xl"
                widget-type="InsightsPageHeaderTitle"
              >
                 
                 
              </h1>
            </Title>
          </PageHeaderTitle>
        </section>
      </PageHeader>
    </Header>
  </Router>
</StaticRouter>
`;

exports[`Header component should render with empty breadcrumb 1`] = `
<StaticRouter
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Header
      breadcrumbs={Array []}
      title=""
    >
      <PageHeader
        data-ouia-component-type="undefined-page-header"
      >
        <section
          className="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
          data-ouia-component-type="undefined-page-header"
          widget-type="InsightsPageHeader"
        >
          <HeaderBreadcrumbs
            items={Array []}
          >
            <Breadcrumb>
              <nav
                aria-label="Breadcrumb"
                className="pf-c-breadcrumb"
                data-ouia-component-id={0}
                data-ouia-component-type="PF4/Breadcrumb"
                data-ouia-safe={true}
              >
                <ol
                  className="pf-c-breadcrumb__list"
                />
              </nav>
            </Breadcrumb>
          </HeaderBreadcrumbs>
          <PageHeaderTitle
            title=""
          >
            <Title
              className=""
              headingLevel="h1"
              size="2xl"
              widget-type="InsightsPageHeaderTitle"
            >
              <h1
                className="pf-c-title pf-m-2xl"
                widget-type="InsightsPageHeaderTitle"
              >
                 
                 
              </h1>
            </Title>
          </PageHeaderTitle>
        </section>
      </PageHeader>
    </Header>
  </Router>
</StaticRouter>
`;

exports[`Header component should render with header Hello world 1`] = `
<StaticRouter
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Header
      title="Hello world"
    >
      <PageHeader
        data-ouia-component-type="undefined-page-header"
      >
        <section
          className="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
          data-ouia-component-type="undefined-page-header"
          widget-type="InsightsPageHeader"
        >
          <PageHeaderTitle
            title="Hello world"
          >
            <Title
              className=""
              headingLevel="h1"
              size="2xl"
              widget-type="InsightsPageHeaderTitle"
            >
              <h1
                className="pf-c-title pf-m-2xl"
                widget-type="InsightsPageHeaderTitle"
              >
                 
                Hello world
                 
              </h1>
            </Title>
          </PageHeaderTitle>
        </section>
      </PageHeader>
    </Header>
  </Router>
</StaticRouter>
`;

exports[`Header component should render with header as empty string 1`] = `
<StaticRouter
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Header
      title=""
    >
      <PageHeader
        data-ouia-component-type="undefined-page-header"
      >
        <section
          className="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
          data-ouia-component-type="undefined-page-header"
          widget-type="InsightsPageHeader"
        >
          <PageHeaderTitle
            title=""
          >
            <Title
              className=""
              headingLevel="h1"
              size="2xl"
              widget-type="InsightsPageHeaderTitle"
            >
              <h1
                className="pf-c-title pf-m-2xl"
                widget-type="InsightsPageHeaderTitle"
              >
                 
                 
              </h1>
            </Title>
          </PageHeaderTitle>
        </section>
      </PageHeader>
    </Header>
  </Router>
</StaticRouter>
`;

exports[`Header component should render without to attribute 1`] = `
<StaticRouter
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Header
      breadcrumbs={
        Array [
          Object {
            "isActive": true,
            "title": "First item",
          },
        ]
      }
      title=""
    >
      <PageHeader
        data-ouia-component-type="undefined-page-header"
      >
        <section
          className="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
          data-ouia-component-type="undefined-page-header"
          widget-type="InsightsPageHeader"
        >
          <HeaderBreadcrumbs
            items={
              Array [
                Object {
                  "isActive": true,
                  "title": "First item",
                },
              ]
            }
          >
            <Breadcrumb>
              <nav
                aria-label="Breadcrumb"
                className="pf-c-breadcrumb"
                data-ouia-component-id={2}
                data-ouia-component-type="PF4/Breadcrumb"
                data-ouia-safe={true}
              >
                <ol
                  className="pf-c-breadcrumb__list"
                >
                  <BreadcrumbItem
                    isActive={true}
                    key=".$First item"
                    showDivider={false}
                  >
                    <li
                      className="pf-c-breadcrumb__item"
                    >
                      First item
                    </li>
                  </BreadcrumbItem>
                </ol>
              </nav>
            </Breadcrumb>
          </HeaderBreadcrumbs>
          <PageHeaderTitle
            title=""
          >
            <Title
              className=""
              headingLevel="h1"
              size="2xl"
              widget-type="InsightsPageHeaderTitle"
            >
              <h1
                className="pf-c-title pf-m-2xl"
                widget-type="InsightsPageHeaderTitle"
              >
                 
                 
              </h1>
            </Title>
          </PageHeaderTitle>
        </section>
      </PageHeader>
    </Header>
  </Router>
</StaticRouter>
`;
